### PR TITLE
fix(flakey-test): refactored getSortedBucketNames for more consistency and predictability. Finished DWP API functionality

### LINKED ISSUE #15798: jstest is failing in circleCI but passes locally
__Steps to reproduce:__

See log when PR checks fail. Here is a screenshot:

image

__Expected behavior:__
Unit tests should pass

__Actual behavior:__
Unit test fails

### PR DESCRIPTION
**Pull request by Asalem1**
Closes #15798 

### Problem

This PR deals with two separate problems:

1. The issue noted above was the result of inconsistent behavior in the test.
2. The API still couldn't handle `!=` deleteWithPredicate conditions

### Solution

1. Refactored the `getSortedBucketNames` to break the data into two separate arrays that are sorted, concatenated and mapped.
2. Connected the `datatypes.ComparisonNotEqual` to the `influxdb.NotEqual` case in `predicate/tag_rule`

### CODE REVIEW

File: ui/src/buckets/selectors/index.ts [hunk 1]
```diff
@@ -5,28 +5,36 @@ export const SYSTEM = 'system'
 
 export const isSystemBucket = (type: string): boolean => type === SYSTEM
 
-export const getSortedBucketNames = (buckets: Bucket[]) =>
-  buckets
-    .sort((a, b) => {
```
Comment (jsteenb2) [inline ui/src/buckets/selectors/index.ts]: funny thing about JS, each browser implements `sort` just a bit different, maddening at times. Wonder if this was passing in some js runtimes and not others 🤔

Comment (Asalem1) [reply ui/src/buckets/selectors/index.ts]: it was actually passing locally but not on circleci. I completely thought that the sort method would work deterministically (especially when it came to two values being compared to one another in a particular order) but I guess not.

:sad_trombone:

File: predicate/tag_rule.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ func NodeComparison(op influxdb.Operator) (datatypes.Node_Comparison, error) {
 	case influxdb.Equal:
 		return datatypes.ComparisonEqual, nil
 	case influxdb.NotEqual:
-		fallthrough
+		return datatypes.ComparisonNotEqual, nil
```
Comment (kelwang) [inline predicate/tag_rule.go]: can you also update the test predicate_test.go?

Comment (Asalem1) [reply predicate/tag_rule.go]: @kelwang definitely

File: ui/src/buckets/selectors/index.ts [hunk 3]
```diff
@@ -5,28 +5,36 @@ export const SYSTEM = 'system'
...
+  buckets.forEach(bucket => {
+    // separate system buckets from the rest
+    if (isSystemBucket(bucket.type)) {
+      systemBuckets.push(bucket)
+    } else {
+      otherBuckets.push(bucket)
+    }
+  })
+  // alphabetize system buckets
+  systemBuckets.sort(sortFunc)
```
Comment (121watts) [inline ui/src/buckets/selectors/index.ts]: All of `sortFunc` could be replaced with 
```
_.sortBy((bucket) => bucket.name.toLocaleLowerCase())
```

File: ui/src/buckets/selectors/index.ts [hunk 4]
```diff
@@ -5,28 +5,36 @@ export const SYSTEM = 'system'
...
+  }
+  if (firstBucket > secondBucket) {
+    return 1
+  }
+  return 0
+}
+
+export const getSortedBucketNames = (buckets: Bucket[]) => {
+  const systemBuckets = []
+  const otherBuckets = []
```
Comment (hoorayimhelping) [inline ui/src/buckets/selectors/index.ts]: minor nit: this should probably be called `userBuckets` but it's not worth blocking on

File: ui/src/buckets/selectors/index.ts (hunk 3 above)
Comment (Asalem1) [reply ui/src/buckets/selectors/index.ts]: I very weirdly can't get it to work using the sortBy function

**Pull request merged**